Assets/GothicVR/Scripts/Creator/WorldCreator.cs

```diff
@@ -30,13 +32,15 @@ public void Create(string worldName)
 
             GameData.I.WorldScene!.Value.GetRootGameObjects().Append(worldGo);
 
-            worldMesh = MeshCreator.I.Create(world, worldGo);
-            VobCreator.I.Create(worldGo, world);
+            worldMesh = await MeshCreator.I.Create(world, worldGo, progress => progressCallback?.Invoke(progress * 0.5f));
```

Comment:
Ah we really need to see how this works. From JavaScript I know, that await is just making an async function synchronous. So no additional benefits. Let's check on it at the end.

Reply:
using await on a function makes the parent function "wait" for our function to finish, but it runs in a non-blocking way for Unity as it should handle everything on the main thread, so instead of blocking the main thread and render the world, we can do async stuff (do small snippets) without blocking the main thread, thus making the loading screen feasible